File: 2020/66.plus-one.js
/*
 * @lc app=leetcode id=66 lang=javascript
 *
 * [66] Plus One
 */

// @lc code=start
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    digits[digits.length-1] += 1
    if (digits[digits.length-1] == 10) {
      let slice = digits.slice(0,-1)
      if (slice.length == 0) slice = [0]
      digits = [...plusOne(slice), ...[0]]
    }
    return digits
};
// @lc code=end

File: 66.plus-one.py
#
# @lc app=leetcode id=66 lang=python3
#
# [66] Plus One
#

# @lc code=start
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        length = len(digits)
        carry = 1
        for i, d in enumerate(digits[::-1]):
            d += carry
            carry = d // 10
            digits[length-1-i] = d % 10
        return ([carry] if carry else []) + digits
# @lc code=end

